app/actions/local/reactions.test.ts

```diff
@@ -0,0 +1,111 @@
+import * as recentReactionsQueries from '@queries/servers/system';
+import * as emojiHelpers from '@utils/emoji/helpers';
+import * as logUtils from '@utils/log';
+
+import {addRecentReaction} from './reactions';
+
+import type ServerDataOperator from '@app/database/operator/server_data_operator';
+
+jest.mock('@database/manager');
+jest.mock('@queries/servers/system');
+jest.mock('@utils/emoji/helpers');
+jest.mock('@utils/log');
+
+describe('addRecentReaction', () => {
+    let operator: ServerDataOperator;
+    const serverUrl = 'baseHandler.test.com';
+
+    beforeEach(async () => {
+        jest.clearAllMocks();
+
+        await DatabaseManager.init([serverUrl]);
+        operator = DatabaseManager.serverDatabases[serverUrl]!.operator;
+
+        (recentReactionsQueries.getRecentReactions as jest.Mock).mockResolvedValue([]);
+        (emojiHelpers.getEmojiFirstAlias as jest.Mock).mockImplementation((emoji) => emoji);
+
+        operator.handleSystem = jest.fn();
+    });
+
+    afterEach(async () => {
+        await DatabaseManager.destroyServerDatabase(serverUrl);
+    });
+
+    it('should return an empty array if emojiNames is empty', async () => {
+        const result = await addRecentReaction(serverUrl, []);
+        expect(result).toEqual([]);
+    });
+
+    it('should add new emoji to the beginning of the list', async () => {
+        const emojiNames = [':air:', ':fire:'];
+        await addRecentReaction(serverUrl, emojiNames);
+
+        expect(operator.handleSystem).toHaveBeenCalledWith({
+            systems: [{
+                id: 'recentReactions',
+                value: JSON.stringify([':fire:', ':air:']),
+            }],
+            prepareRecordsOnly: false,
+        });
+    });
+
+    it('should move existing emoji to the beginning of the list', async () => {
+        (recentReactionsQueries.getRecentReactions as jest.Mock).mockResolvedValue([':fire:', 'water']);
+        const emojiNames = [':air:', ':fire:'];
+        await addRecentReaction(serverUrl, emojiNames);
+
+        expect(operator.handleSystem).toHaveBeenCalledWith({
+            systems: [{
+                id: 'recentReactions',
+                value: JSON.stringify([':fire:', ':air:', 'water']),
+            }],
+            prepareRecordsOnly: false,
+        });
+    });
+
+    it('should limit the list to MAXIMUM_RECENT_EMOJI', async () => {
+        const longEmojiList = Array.from({length: 40}, (_, i) => `emoji${i}`);
+        (recentReactionsQueries.getRecentReactions as jest.Mock).mockResolvedValue(longEmojiList);
+        await addRecentReaction(serverUrl, ['newEmoji']);
+
+        const handleSystemCall = (operator.handleSystem as jest.Mock).mock.calls[0][0];
+        const savedEmojis = JSON.parse(handleSystemCall.systems[0].value);
+        expect(savedEmojis.length).toBe(27);
+        expect(savedEmojis[0]).toBe('newEmoji');
+    });
+
+    it('should use getEmojiFirstAlias for each emoji', async () => {
```

Comment:
The idea here is to use a emoji that have several aliases, and see which alias is stored, the one passed, or the first alias. It should store the first alias, and not whatever has been passed.

Reply:
@larkox , 
for example:

if stored alias of Air is Breeze, and fire is Flame

and I pass [Breeze, flame] the output should be [Air, fire] ??

Reply:
Yes, that is the implemented behavior.